eli5/catboost.py

```diff
@@ -0,0 +1,90 @@
+
+from __future__ import absolute_import, division
+
+import numpy as np  # type: ignore
+import catboost # type: ignore
+
+from eli5.explain import explain_weights
+from eli5._feature_importances import get_feature_importance_explanation
+
+DESCRIPTION_CATBOOST = """
+CatBoost feature importances; values are numbers 0 <= x <= 1;
+all values sum to 1.
+"""
+
+
+@explain_weights.register(catboost.CatBoost)
+@explain_weights.register(catboost.CatBoostClassifier)
+@explain_weights.register(catboost.CatBoostRegressor)
+def explain_weights_catboost(catb,
+                             top=20,
+                             importance_type='PredictionValuesChange',
+                             prettified=True,
+                             feature_names=None,
+                             pool=None
+                             ):
+    """
+    Return an explanation of an CatBoost estimator (CatBoostClassifier, CatBoost, CatBoostRegressor)
+    as feature importances.
+
+    See :func:`eli5.explain_weights` for description of
+    ``top``, ``feature_names``,
+    ``feature_re`` and ``feature_filter`` parameters.
+
+    ``target_names`` and ``targets`` parameters are ignored.
+
+    Parameters
+    ----------
+    importance_type : str, optional
+        A way to get feature importance. Possible values are:
+        - 'PredictionValuesChange' - The individual importance values for each of the input features.
+          (default)
+        - 'LossFunctionChange' - The individual importance values for each of the input features for ranking metrics (requires training data to be passed  or a similar dataset with Pool)
+
+    prettified : bool, optional
+        A way to get the feature names from the columns and display it against the feature importance
+        - 'True' - return list of tuples on get_feature_importance on catboost objects along with the column names. (default)
+        - 'False' - return list numbers on get_feature_importance on catboost objects.
+
+    pool : catboost.Pool, optional
+        To be passed if explain_weights_catboost has importance_type set to "LossFunctionChange".
+        The catboost feature_importances_ uses the Pool datatype to calculate the parameter for the specific importance_type.
+
+    """
+    is_regression = _check_catboost_args(catb)
+    catb_feature_names = catb.feature_names_
+    coef = _catb_feature_importance(catb, importance_type=importance_type, pool=pool)
+    #vec is only added since it is a positional argument in get feature_importance_explanation and cannot 
+    #be made into a keyword argument since it is followed by a positional argument.
+    vec = None
```

Comment:
Sorry for creating confusion regarding the ``vec`` variable - I think it would be nice to have it as a keyword argument, because having some vectoriser before CatBoost can still make sense - e.g. this can be vectoriser transforming words into features, so explanation would be given in terms of words and not feature names. So I suggest bringing it back into function arguments, with ``None`` default - sorry again that I was not clear here.

Reply:
okay will do